client/oauth.py

```
@@ -0,0 +1,316 @@
+    transport = session or requests
+    response = transport.post(
+        token_endpoint,
+        verify=REQUESTS_VERIFY_SSL,
+        data=payload,
+        headers={"Content-Type": "application/x-www-form-urlencoded", "Cache-Control": "no-cache"},
+        timeout=10,
+    )
+    response.raise_for_status()
+    token_data = response.json()
+    if not isinstance(token_data, Mapping):
+        raise RuntimeError("Token endpoint returned non-JSON response")
```

Ruff (0.14.1)

Prefer `TypeError` exception for invalid type

(TRY004)